File: react-native/react-native-totvs-clockin-vision/android/src/main/java/com/totvs/clockin/vision/events/OnFaceProximity.kt
package com.totvs.clockin.vision.events

import com.facebook.react.bridge.Arguments
import com.facebook.react.bridge.ReactContext
import com.facebook.react.uimanager.events.RCTEventEmitter

/**
 * Event emitted when the user face is too small to be eligible for
 * face recognition task.
 */
data class OnFaceProximity(
    val isUnderThreshold: Boolean = false,
    val threshold: Float,
    val faceWidth: Float,
    val faceHeight: Float
) : Event {

    override fun invoke(context: ReactContext, viewId: Int) {
        val event = Arguments.createMap().apply {
            putBoolean(FIELD_IS_UNDER_THRESHOLD, isUnderThreshold)
            putDouble(FIELD_THRESHOLD, threshold.toDouble())
            putDouble(FIELD_FACE_WIDTH, faceWidth.toDouble())
            putDouble(FIELD_FACE_HEIGHT, faceHeight.toDouble())
        }

        context.getJSModule(RCTEventEmitter::class.java).receiveEvent(viewId, name, event)
    }

    companion object : Event.Export {
        override val name = "onFaceProximity"

        private const val FIELD_IS_UNDER_THRESHOLD = "isUnderThreshold"
        private const val FIELD_THRESHOLD = "threshold"
        private const val FIELD_FACE_WIDTH = "faceWidth"
        private const val FIELD_FACE_HEIGHT = "faceHeight"
    }
}

File: react-native/react-native-clockin-vision/ios/Events/OnFaceProximity.swift
//
//  OnFaceProximity.swift
//  TOTVSClockInVision
//
//  Created by Jansel Rodriguez on 8/3/20.
//  Copyright © 2020 Facebook. All rights reserved.
//

import Foundation

fileprivate let FIELD_IS_UNDER_THRESHOLD = "isUnderThreshold"
fileprivate let FIELD_THRESHOLD = "threshold"
fileprivate let FIELD_FACE_WIDTH = "faceWidth"
fileprivate let FIELD_FACE_HEIGHT = "faceHeight"

/**
* Event emitted when the user face is too small to be eligible for
* face recognition task.
*/
public class OnFaceProximity : Event {
        
    private let emit: RCTDirectEventBlock?
    
    public required init(emit: RCTDirectEventBlock?) {
        self.emit = emit
    }
    
    public func send(data: ProximityResult) {
        let event: [String: Any] = [
            FIELD_IS_UNDER_THRESHOLD: data.isUnderThreshold,
            FIELD_THRESHOLD: data.threshold,
            FIELD_FACE_WIDTH: data.faceWidth,
            FIELD_FACE_HEIGHT: data.faceHeight
        ]
        emit?(event)
    }
}
